skimage/registration/_phase_cross_correlation.py

```diff
@@ -190,6 +190,13 @@ def phase_cross_correlation(reference_image, moving_image, *,
         return _masked_phase_cross_correlation(reference_image, moving_image,
                                                reference_mask, moving_mask,
                                                overlap_ratio)
+    # If input images contain NaN pixels, must mask before correlation
+    if np.isnan(reference_image).any() or np.isnan(moving_image).any():
```

Comment:
`np.isnan` is called twice  on both images: once for checking and the other for creating the mask.
May be the mask can be create before checking to avoid this repetition...

Reply:
Python 3.8 syntax would be so nice for this:

```python
if (reference_mask := np.isnan(reference_image)).any() ...:
```

Just doing some advertising for the future. =)